src/Provider/AbstractProvider.php

```diff
@@ -622,10 +622,10 @@ protected function fetchUserDetails(AccessToken $token)
      *
      * Typically this is used to set Accept or Content-Type headers.
      *
-     * @param  AccessToken $token
+     * @param  AccessToken|null $token
      * @return array
      */
-    protected function getDefaultHeaders($token = null)
+    protected function getDefaultHeaders(AccessToken $token = null)
```

Comment:
As per my comments on #359, this method shouldn't even have a `$token` parameter.